lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp

```diff
@@ -610,34 +693,51 @@ namespace Js
         }
 
         return JavascriptString::NewCopySz(wgl->WindowsGetStringRawBuffer(*locale, NULL), scriptContext);
+#endif
     }
 
     Var IntlEngineInterfaceExtensionObject::EntryIntl_GetDefaultLocale(RecyclableObject* function, CallInfo callInfo, ...)
     {
         EngineInterfaceObject_CommonFunctionProlog(function, callInfo);
 
-        WCHAR defaultLocale[LOCALE_NAME_MAX_LENGTH];
+        char16 defaultLocale[LOCALE_NAME_MAX_LENGTH];
         defaultLocale[0] = '\0';
 
-        if (GetUserDefaultLocaleName(defaultLocale, _countof(defaultLocale)) == 0 || defaultLocale[0] == '\0')
+        if (
+#if defined(INTL_WINGLOB)
+            // TODO (doilij) investigate why the PAL definition of this function wasn't found in Linux build.
+            GetUserDefaultLocaleName(defaultLocale, _countof(defaultLocale)) == 0
+#else
+            IcuIntlAdapter::GetUserDefaultLocaleName(defaultLocale, _countof(defaultLocale)) == 0
+#endif
+            || defaultLocale[0] == '\0')
         {
+#if defined(INTL_WINGLOB)
             return scriptContext->GetLibrary()->GetUndefined();
+#else
+            Output::Print(_u("EntryIntl_GetDefaultLocale > IcuIntlAdapter::GetUserDefaultLocaleName returned 0: returning null to fallback to JS\n"));
+
+            return scriptContext->GetLibrary()->GetNull();
```

Comment:
It looks like there are a lot of cases in this file where for the non-`WINGLOB` case you return null on errors, while the Winglob code returns Undefined instead. Is that deliberate?